Code block: Enhance core instead of new block

sirreal opened the pull request:
## Proposed changes:

Update the `a8c/code` block to enhance the `core/code` block instead of providing an independent block.

### Known issues:

The known issue has been resolved.

~The global styles system doesn't appear to recognize the block. For some reason, it does not add styles as expected. A code block with in default twentytwentyfive has a font family, but this is never added to the inline styles when the block override is in place:~

```css
:root :where(.wp-block-code){
  font-family: var(--wp--preset--font-family--fira-code);
}
```

~Custom block `fontFamily` is currently disabled. Due to this global styles issue and other complexities, `fontFamily` support is not available but should be added at some point.~


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

The block is not enabled and will have thorough testing before release.

[Testing setup instructions are here.](https://github.com/Automattic/jetpack/pull/45389#issuecomment-3373254266) It's available on sites with the `gutenberg-edge` flag for testing. The most important thing to verify is that the feature is not enabled by default. That logic has not changed in this PR.

sirreal committed:
Remove unused class constants

sirreal committed:
First pass change from new block to core/code mod

sirreal committed:
Seems to be working

sirreal committed:
changelog

sirreal committed:
Update/restore types

sirreal committed:
cleanup

sirreal committed:
Fix transform settings

sirreal committed:
Redo transforms for compatibility with core/code

sirreal committed:
Revert "Redo transforms for compatibility with core/code"

This reverts commit 7f6f0670aff8f282797ea13c94a56c055fbcf956.

sirreal committed:
Ensure transforms are working correctly

sirreal committed:
Switch to "rich text" content instead of "code" string atribute

sirreal committed:
Fix double tag name output

sirreal committed:
Improve tokenizedLines type and use it in construction

sirreal committed:
Remove unused example block

sirreal committed:
Remove codeblocksettings interface

sirreal committed:
Enable and improve types for syntaxhighlighter/code transform filter

sirreal committed:
Remove debug logging

sirreal committed:
Add jsdoc param description

sirreal committed:
Remove commented transforms

sirreal committed:
Clean up code block class lints

sirreal committed:
Clean up stylesheet

sirreal committed:
Fix RichTextData usage so transforms work correctly.

sirreal committed:
Fix imported types

sirreal committed:
Fix an improve files transform to use content

sirreal committed:
Remove block.json

sirreal committed:
Update to use `core/code` custom theme colors

sirreal committed:
Remove unused styles and simplify

sirreal committed:
Drop fontFamily support

sirreal committed:
Remove commented code

sirreal committed:
Apply stylelint fixes

sirreal committed:
Use wp-rich-text version already in use

dmsnell reviewed (dismissed):
Having looked over this, nothing jumped out, other than I liked how easy it was to read the change in HTML-processing logic.

I’m trusting you tested this, because I only reviewed the code itself.

What happens if we use this block and it replaces the Core code block, and then this plugin disappears? Are the blocks easy and reliable to transform back into Core code blocks?

sirreal committed:
Fix problem with global styles system

sirreal committed:
Update how style src is changed

sirreal committed:
Avoid setting font family

sirreal committed:
Improve HTML cleanup, remove PRE tag

sirreal committed:
Restore the PRE tag

sirreal committed:
Use global NS for functions

sirreal committed:
Handle font family well

This supports UA styling, theme styling of PRE elements, theme.json and global styles font-family styling.

sirreal committed:
Resolve font-family and typography support issues

sirreal committed:
update lockfile

sirreal committed:
Move language name class to dynamic render to avoid block recovery

sirreal committed:
Override more styles to prevent code element from being styled inside pre element

sirreal committed:
Fix phan offset warning

sirreal committed:
Fix typo and improve comment

sirreal committed:
Fix typo in comment

sirreal committed:
Remove comment

sirreal committed:
Fix to use init action instead of filter

sirreal committed:
Add some safety to content.text access

sirreal committed:
Fix $html param description

sirreal committed:
Fix typo in comment

sirreal committed:
Tweak some style spacing to better align content and match core block
